File: backend/service.py
import httpx

API_KEY = "demo"  # replace later with real token

async def fetch_aqi_data(city: str):
    url = f"https://api.waqi.info/feed/{city}/?token={API_KEY}"

    async with httpx.AsyncClient() as client:
        response = await client.get(url)

    data = response.json()

    if data.get("status") != "ok":
        return {"city": city, "aqi": None, "error": "City not found"}

    d = data["data"]

    return {
        "city": d["city"]["name"],
        "aqi": d["aqi"],
        "updated": d["time"]["s"],
    }

File: backend/service.js
import axios from "axios";

const API_TOKEN = "demo"; // later replace with real token

export async function fetchAQI(city) {
  const url = `https://api.waqi.info/feed/${city}/?token=${API_TOKEN}`;

  const response = await axios.get(url);
  const data = response.data;

  if (data.status !== "ok") {
    return { city, aqi: null, error: "City not found" };
  }

  return {
    city: data.data.city.name,
    aqi: data.data.aqi,
    dominantPollutant: data.data.dominentpol,
    updated: data.data.time.s
  };
}
